### ISSUE TITLE
Locally installed version of drush is messing with our workflow

### ISSUE THREAD

**Issue by bmcclure** [kalabox/kalabox#1371]
Bug Report
==========

**Tell us about your setup**

What is your Kalabox version and operating system?

v0.12.0-beta1 on Windows 10 Professional 64-bit

Have you previously installed Kalabox? If so, what version?

Yes, all released alpha versions.

**Tell us about your bug**

Give a brief description of what happened, including any error messages that appeared.

During the "Pulling database" phase of creating a new project with the CLI, I receive the following error:

```
PHP Warning:  pcntl_exec(): Error has occurred: (errno 13) Permission denied in phar:///usr/local/bin/drush/includes/startup.inc on line 359

Warning: pcntl_exec(): Error has occurred: (errno 13) Permission denied in phar:///usr/local/bin/drush/includes/startup.inc on line 359
Error has occurred executing the Drush script found at /code/vendor/bin/drush.launcher
(errno 13) Permission denied
Pulling files.
```

Please detail the steps we can take to replicate this bug. Be as detailed as possible.

> Example: 

1. Open command prompt
2. Browse to location where new project should be created
3. Run "kbox create pantheon" to clone the correct project and environment
4. Wait for "pulling database step" and error shows up

**Tell us more**

Could better documentation have prevented this issue? if so what should we change?

Does this bug prevent you from using Kalabox?

Yes, I haven't been able to get a site to create successfully yet.

**Comment by pirog** [comment]
@bmcclure can you tell us more about what your `drush.launcher` looks like? 
Are you using the normal kalabox drush or something else?

**Comment by amklose** [comment]
I'm getting the same errors on a Mac OS X 10.11.5...

```
Pulling code.
error: cannot open .git/FETCH_HEAD: Permission denied

Pulling database.
PHP Warning:  pcntl_exec(): Error has occurred: (errno 13) Permission denied in phar:///usr/local/bin/drush/includes/startup.inc on line 359

Warning: pcntl_exec(): Error has occurred: (errno 13) Permission denied in phar:///usr/local/bin/drush/includes/startup.inc on line 359
Error has occurred executing the Drush script found at /code/vendor/bin/drush.launcher
(errno 13) Permission denied
Pulling files.
Warning: Permanently added '[appserver.tony.2a745e15-4221-4b40-aa9c-fd33e2959304.drush.in]:2222,[192.237.244.56]:2222' (RSA) to the list of known hosts.
receiving incremental file list

sent 146 bytes  received 606 bytes  167.11 bytes/sec
total size is 88,685  speedup is 117.93
PHP Warning:  pcntl_exec(): Error has occurred: (errno 13) Permission denied in phar:///usr/local/bin/drush/includes/startup.inc on line 359

Warning: pcntl_exec(): Error has occurred: (errno 13) Permission denied in phar:///usr/local/bin/drush/includes/startup.inc on line 359
Error has occurred executing the Drush script found at /code/vendor/bin/drush.launcher
(errno 13) Permission denied

```

**Comment by bmcclure** [comment]
@pirog We are using a site-local copy of Drush installed via Composer on the project, similar to what's described here: https://pantheon.io/blog/avoiding-dependency-hell-site-local-drush

Is this causing a conflict with kalabox do you think?

**Comment by pirog** [comment]
That would be my guess. If you can detail your setup a little more i can try to replicate this.

**Comment by bmcclure** [comment]
@pirog Here's a Gist that contains our composer.json file. Every bit of code on the site so far was either included with Pantheon's D8 distribution, or added by this composer.json file, so one way to reproduce this would be to create a new D8 site on Pantheon, use my composer.json to install all dependencies, and then try creating a local dev environment of that using Kalabox.

This error occurs both on a Mac and Windows workstation, so I don't think it has anything to do with my OS. Thanks for your assistance!

**Comment by pirog** [comment]
Thanks @bmcclure, im almost positive that this is just a weird issue between using a local drush bin and our app specific one. hoping once i can replicate this it will be easy to resolve.

**Comment by hardlyneutral** [comment]
Has there been any movement on this @pirog? I'm running into an identical issue with several of my sites.

**Comment by pirog** [comment]
Haven't had a chance to try and replicate yet.

**Comment by amklose** [comment]
I just tried again today with a fresh installation of Kalabox and got the same error...

**Comment by pirog** [comment]
We haven't released a new version since the beta1 so that is unsurprising.

**Comment by pirog** [comment]
@amklose looping back around on this. Does your site have `composer.json` that installs drush? Any chance you can add me to the site that is failing?

**Comment by bmcclure** [comment]
I believe in my case the issue was that I installed my composer dependencies on Windows and the executable files got saved without the executable bit, causing a lot of weird errors with Drush until I installed the dependencies from a Linux machine and committed them to the repo that way. Likely a side effect of having to commit the composer dependencies to the repo, and something with my version of git or cygwin or something must be causing an issue since I'm running on Windows.

As far as I'm concerned this is working fully now, since installing the dependencies (including the site-local drush and its binaries) on a Linux box and committing them from there, I haven't had an issue on the last couple of versions of kalabox with this specific error.

Unsure if @hardlyneutral is still experiencing a similar problem, or if his cause was similar to mine or not however.
